Added available sort options for all REST API endpoints that allow sorting

monotasker opened the pull request:
:heart: Thank you for your contribution!

### Description

Added documentation on the sort options available for all REST API endpoints that allow sorting.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [X ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [X ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [X ] I've added relevant test cases.
- [X ] I've added relevant documentation.
- [X ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/) (for relevant code).
- [X ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [X ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [X ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [X ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [X ] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

monotasker commented on docs/reference/rest_api_vocabularies.md:
```diff
@@ -33,6 +33,20 @@ Specifically for the `application/vnd.inveniordm.v1+json` format:
 
 The API uses a locale matching algorithm, that will do its best effort to translate the vocabulary record's title and description.
 
+The sort options available vary depending on the vocabulary being searched. These cannot presently be configured via config variables but are set in the service configuration for each vocabulary service. The default sort options are used where a vocabulary service has not defined its own:
+
+| Vocabulary | Available sort options | Default | Default without a query string |
+| ---------- | ---------------------- | ------- | ------------------------------ |
+| default | `"bestmatch"`, `"title"`, `"newest"`, `"oldest"` | `"bestmatch"` | `"title"` |
+| affiliations | `"bestmatch"`, `"name"`, `"newest"`, `"oldest"` | `"bestmatch"` | `"name"` |
+| awards | `"bestmatch"`, `"title"`, `"newest"`, `"oldest"` | `"bestmatch"` | `"title"` |
+| funders | `"bestmatch"`, `"name"`, `"newest"`, `"oldest"` | `"bestmatch"` | `"name"` |
+| names | `"bestmatch"`, `"name"`, `"newest"`, `"oldest"` | `"bestmatch"` | `"name"` |
+| subjects* | `"bestmatch"`, `"subject"`, `"newest"`, `"oldest"` | `"bestmatch"` | `"subject"` |
+
+(\* Subjects are available only at the `/api/subjects` endpoint, but are included here for reference since they are provided by a vocabulary service.)
+<!--Maybe the subjects sort options should be moved to documentation for the `api/subjects` endpoint-->
```
Makes sense. I'll make sure the order is flipped on all the endpoints. Then I'll remove the table and just document the default sort options. I think it's still useful to have the defaults documented, since not all vocabularies have their own endpoint.

fenekku commented on docs/reference/rest_api_communities.md:
```diff
@@ -293,12 +293,13 @@ Content-Type: application/json
 | `type`   | string  | query    | Specify community type as one of organization, event, topic or project.                                                                                                                                    |
 | `accept` | string  | header   | - `application/json` (default)<br />- `application/vnd.inveniordm.v1+json`                                                                                                                                 |
 
+Sort options for communities can be configured using the `COMMUNITIES_SORT_OPTIONS` config variable as described in the [search customization](../customize/search) section. Note that `"bestmatch"` is only available as a sort option on requests that provide a query string as a `q` parameter. Otherwise `"bestmatch"` is ignored and the default `"newest"` sort is used.
```
The failing tests are because the links have to be to the markdown files. Mkdocs then takes care of converting them to regular files. So we have to replace:

```suggestion
Sort options for communities can be configured using the `COMMUNITIES_SORT_OPTIONS` config variable as described in the [search customization](../customize/search.md) section. Note that `"bestmatch"` is only available as a sort option on requests that provide a query string as a `q` parameter. Otherwise `"bestmatch"` is ignored and the default `"newest"` sort is used.
```

and then we should be fine :crossed_fingers: .

monotasker replied:
Ah, right. Thanks

fenekku reviewed (commented):
Just the failing tests to fix and we are good to merge :smile:

monotasker commented:
Okay, I've made the minor changes to default option ordering, removed the table of vocab options (leaving a description of the defaults), and fixed the missing file extensions in the links. Should be good to go.

fenekku commented:
Hmm something is off with the links (see https://github.com/inveniosoftware/docs-invenio-rdm/actions/runs/8582061590/job/23563183805?pr=612), but at least one other PR is potentially affected: https://github.com/inveniosoftware/docs-invenio-rdm/actions/runs/8526851030/job/23356983728?pr=614 ... I will add this as something I will look into this month as part of preparations for v12 since we need to make sure docs are good.

monotasker commented:
Okay, I rebased my branch but it doesn't seem to have fixed the issue. When I build the docs locally I get the same error: not from links in the documentation text but because of two missing style files: /assets/stylesheets/main.10ba22f1.min.css and /assets/stylesheets/palette.06af60db.min.css. These are linked in the frontpage_base.html template, and the files do indeed seem to be missing. I don't think this is related to any of my changes. Maybe something happened upstream that is just becoming visible here? (Or maybe I'm missing something!)

fenekku commented:
Indeed something is strange. If you don't mind, when I look into it, may I push to your branch to fix it? This will be faster and you've already done the heavy lifting, so you shouldn't have to deal with that structural problem :wink:  .

monotasker commented:
Sure. No problem

fenekku commented:
Merged it here: https://github.com/inveniosoftware/docs-invenio-rdm/pull/619 (@monotasker 's master branch is probably protected so I couldn't push to it). I think Nico fixed the issue by resetting the clean options: https://github.com/inveniosoftware/docs-invenio-rdm/commit/c6e7334fe0a9386d1db42fe499cd5c53947be556#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fR41 . Kind of strange nevertheless.
